src/Testcontainers/Configurations/Modules/Databases/CosmosDbTestcontainerConfiguration.cs

```diff
@@ -0,0 +1,67 @@
+namespace DotNet.Testcontainers.Configurations
+{
+    using DotNet.Testcontainers.Builders;
+    using JetBrains.Annotations;
+
+    [PublicAPI]
+    public class CosmosDbTestcontainerConfiguration : TestcontainerDatabaseConfiguration
+    {
+        private const string CosmosDbImage = 
+            "mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator";
+
+        private const int CosmosDbPort = 8081;
+
+        private const string CosmosDbDefaultKey =
+            "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==";
+
+        private const string CosmosDbDefaultAccountName = 
+            "localhost:8081";
+
+        public CosmosDbTestcontainerConfiguration()
+            : this(CosmosDbImage) { }
+
+        public CosmosDbTestcontainerConfiguration(string image)
+            : base(image, CosmosDbPort, CosmosDbPort)
+        {
+          this.Environments.Add("AZURE_COSMOS_EMULATOR_PARTITION_COUNT", "1");
+        }
+
+        public override string Username
+        {
+            get => CosmosDbDefaultAccountName;
+        }
+
+        public override string Password
+        {
+            get => CosmosDbDefaultKey;
+        }
+
+        public override string Database
+        {
+            get => this.Environments["AZURE_COSMOS_EMULATOR_DATABASE"];
+            set => this.Environments["AZURE_COSMOS_EMULATOR_DATABASE"] = value;
+        }
+
+        public string PartitionCount
+        {
+            get => this.Environments["AZURE_COSMOS_EMULATOR_PARTITION_COUNT"];
+            set => this.Environments["AZURE_COSMOS_EMULATOR_PARTITION_COUNT"] = value;
+        }
+
+        public string EnableDataPersistence
+        {
+            get => this.Environments["AZURE_COSMOS_EMULATOR_ENABLE_DATA_PERSISTENCE"];
+            set => this.Environments["AZURE_COSMOS_EMULATOR_ENABLE_DATA_PERSISTENCE"] = value;
+        }
+
+        public string IPAddressOverride
+        {
+            get => this.Environments["AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE"];
+            set => this.Environments["AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE"] = value;
+        }
+
+        public override IWaitForContainerOS WaitStrategy => Wait.ForUnixContainer()
+            .UntilCommandIsCompleted("ls");
+            //.UntilMessageIsLogged(this.OutputConsumer.Stdout, "Started");
```

Comment:
You probably need to read `stdout`, here is an example: https://github.com/testcontainers/testcontainers-dotnet/discussions/438#discussion-3949974.

Reply:
Hmm, I can't get this to work. The stdout wait strategy deadlocks for me, debugging in the wait strategy itself shows output remains empty. So I'm guessing the output stream is not coming through. Might be because I'm running linux containers on docker for windows atm?

Any ideas?

Reply:
See below